services/communitytokens/service.go

```diff
@@ -442,6 +451,33 @@ func (s *Service) DeploymentSignatureDigest(chainID uint64, addressFrom string,
 	return s.manager.DeploymentSignatureDigest(chainID, addressFrom, communityID)
 }
 
+func (s *Service) ProcessCommunityTokenAction(message *protobuf.CommunityTokenAction) error {
+	communityToken, err := s.Messenger.GetCommunityTokenByChainAndAddress(int(message.ChainId), message.ContractAddress)
+	if err != nil {
+		return err
+	}
+	if communityToken == nil {
+		return fmt.Errorf("can't find community token in database: chain %v, address %v", message.ChainId, message.ContractAddress)
+	}
+
+	if message.ActionType == protobuf.CommunityTokenAction_BURN {
```

Comment:
No need, we upgrade database only in case of burn (because token has different supply).
In case of remote destruct and airdrop we only need to refresh token holders.